src/admin/items/items.service.ts

```diff
@@ -258,63 +259,80 @@ export class ItemsService {
 		);
 	}
 
-	public static async fetchItemByExternalId(
-		externalId: string
-	): Promise<(Avo.Item.Item & { replacement_for?: string }) | null> {
+	public static async fetchItemsByExternalId(
+		externalIds: string[]
+	): Promise<Array<(Avo.Item.Item & { replacement_for?: string }) | null>> {
```

Comment:
null should not be a possible response type here, since you always return at least an empty array.
If you do sometimes return null, you need to handle it correctly in the code that uses this function.

Reply:
Items in the array can be `null`, not the entire response.
`Promise<Array<... | null>>`